### PR TITLE
feat(plasma-hope,plasma-web,plama-b2c): Add `Button` union 

### PR DESCRIPTION
**Pull request by neretin-trike**
Необходимо было объединить компонент `Button`. Теперь в библиотеках plasma-web и plasma-b2c реализация этого компонента и API (т.е. все его свойства) - одинаковая. 
В самом plasma-hope основная логика выделена в файл `Button`, в котором в зависимости от выбранного свойства `design`, будет возвращаться вью компонента.

Итоговая структура компонента:
```
plasma-web // directory
|
└─ Button // re-export with `design`
  |
  └─ plasma-hope //  directory
    |
    └─ Button // business logic
       |
       └─ ButtonB2C/Web // view
```

:baby_chick: Download canary assets:

[colors--canary.453.4551135183.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.453.4551135183.xml)  
[color_metro_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.453.4551135183.swift)  
[color_metro_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.453.4551135183.kt)  
[color_metro_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.453.4551135183.ts)  
[color_sberHealth_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.453.4551135183.swift)  
[color_sberHealth_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.453.4551135183.kt)  
[color_sberHealth_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.453.4551135183.ts)  
[color_sbermarket_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.453.4551135183.swift)  
[color_sbermarket_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.453.4551135183.kt)  
[color_sbermarket_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.453.4551135183.ts)  
[color_sberprime_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.453.4551135183.swift)  
[color_sberprime_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.453.4551135183.kt)  
[color_sberprime_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.453.4551135183.ts)  
[color_selgros_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.453.4551135183.swift)  
[color_selgros_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.453.4551135183.kt)  
[color_selgros_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.453.4551135183.ts)  
[color_smbusiness_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.453.4551135183.swift)  
[color_smbusiness_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.453.4551135183.kt)  
[color_smbusiness_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.453.4551135183.ts)  
[PlasmaTokensColor--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.453.4551135183.swift)  
[shadow_sbermarket_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.453.4551135183.ts)  
[typo_mage_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.453.4551135183.swift)  
[typo_mage_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.453.4551135183.kt)  
[typo_mage_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.453.4551135183.ts)  
[typo_plasma_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.453.4551135183.swift)  
[typo_plasma_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.453.4551135183.kt)  
[typo_plasma_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.453.4551135183.ts)  
[typo_ruler_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.453.4551135183.swift)  
[typo_ruler_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.453.4551135183.kt)  
[typo_ruler_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.453.4551135183.ts)  
[typo_sage_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.453.4551135183.swift)  
[typo_sage_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.453.4551135183.kt)  
[typo_sage_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.453.4551135183.ts)  
[typo_sbermarket_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.453.4551135183.swift)  
[typo_sbermarket_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.453.4551135183.kt)  
[typo_sbermarket_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.453.4551135183.ts)  
[typo_soulmate_ios-swift--canary.453.4551135183.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.453.4551135183.swift)  
[typo_soulmate_kotlin--canary.453.4551135183.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.453.4551135183.kt)  
[typo_soulmate_react-native--canary.453.4551135183.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.453.4551135183.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.169.0-canary.453.4551135183.0
  npm install @salutejs/plasma-hope@0.27.0-canary.453.4551135183.0
  npm install @salutejs/plasma-web@1.194.0-canary.453.4551135183.0
  # or 
  yarn add @salutejs/plasma-b2c@1.169.0-canary.453.4551135183.0
  yarn add @salutejs/plasma-hope@0.27.0-canary.453.4551135183.0
  yarn add @salutejs/plasma-web@1.194.0-canary.453.4551135183.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-453/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-453/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-453/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-453/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-453/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-453/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-453/temple-storybook/

File: packages/plasma-hope/src/components/Modal/ModalView.tsx [hunk 1]
```diff
@@ -61,6 +61,7 @@ export const ModalView = React.forwardRef<HTMLDivElement, ModalViewProps>(
                 <StyledBody {...rest} ref={ref} role={role} aria-modal="true">
                     <StyledContent>{children}</StyledContent>
                     <StyledButtonClose
+                        design="web"
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Modal/ModalView.tsx:64]: а почему так?

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: packages/plasma-hope/src/components/Modal/ModalView.tsx (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-hope/src/components/Modal/ModalView.tsx:64]: решили пока оставить так до глобального редизайна

**Review by Yakutoc — APPROVED** [review]

**Pull request merged**